fraud/paypaldatacollectorutil.md

```diff
@@ -0,0 +1,134 @@
+| :-------------------- | :--------------------------------------------------------------------------------------------- | :------------------------------------------------ | :----------------- |
+| config                |  The configuration for PayPal Data Collector util.                                             | `PayPalDataCollectorConfig`                       | Mandatory          |
+
+#### PayPalDataCollectorConfig
+
+This subsection describes the various parameters required by the `PayPalDataCollectorConfig`.
+
+The following sample code demonstrates the response structure:
+
+```Kotlin
+data class PayPalDataCollectorConfig(
+    val accessToken: String,
+    val gatewayId: String
+)
+```
+
+| Name           | Definition                                                                       | Type                       | Mandatory/Optional |
+| :------------- | :------------------------------------------------------------------------------- | :------------------------- | :----------------- |
+| accessToken    |  The OAuth access token required for authenticating API requests.                | String                     | Mandatory          |
+| gatewayId      |  The PayPal gateway ID used to identify the payment gateway.                     | String                     | Mandatory          |
+
+The following sample code demonstrates the `collectDeviceInfo()` usage and params:
+
+```Kotlin
+fun collectDeviceInfo(
+        context: Context,
+        hasUserLocationConsent: Boolean = false,
+        clientMetadataId: String? = null,
+        additionalData: Map<String, String>? = null,
+    ): String? {...}
+```
+
+```Kotlin
+val clientMetadataId = payPalDataCollectorUtil?.collectDeviceInfo(context)
+```
+
+> **Note**:
+>
+> All parameters are optional fields besides the `context` which mirrors the PayPal SDK `PayPalDataCollectorRequest`.
+
+| Name                      | Definition                                                                                                                                                     | Type                                       | Mandatory/Optional |
+| :------------------------ | :------------------------------------------------------------------------------------------------------------------------------------------------------------- | :----------------------------------------- | :----------------- |
+| context                   |  The OAuth access token required for authenticating API requests.                                                                                              | `Context`                                  | Mandatory          |
+| hasUserLocationConsent    |  Informs the SDK if your application has obtained consent from the user to collect location data in compliance with Google Play Developer Program policies.    | Boolean (default = false)                  | Optional           |
```

Comment:
@RicardoDS-Paydock Is there any extra info from PayPal on this field in terms of the benefits of using it?

Reply:
Not that i could find, the definition is what is mentioned from the PayPal SDK. From what I have seen is that if your app requests location permissions it can be used here. There has been issues in the past that this SDK caused vialoations due to automatically tracking location which now seems to have moved to a param type request.

Reply:
This is what i found on their docs:

User Data Consent
Merchant applications are responsible for collecting user data consent. If your app has obtained consent from the user to collect location data in compliance with [Google Play Developer Program policies](https://support.google.com/googleplay/android-developer/answer/10144311#personal-sensitive), set hasUserLocationConsent to true. This flag enables PayPal to collect necessary information required for Fraud Detection and Risk Management.

Merchant App Disclosure
Merchant applications may be required to display a disclosure before collecting user location data in accordance with Google’s [Best practices for prominent disclosures and consent](https://support.google.com/googleplay/android-developer/answer/11150561?hl=en&ref_topic=12797379&sjid=10421482417907285178-NC). By setting hasUserLocationConsent to true, your app is enabled to share device location data with a third party (PayPal) for Fraud Detection and Risk Management.

Reply:
@RicardoDS-Paydock Cool, just add that "This flag enables PayPal to collect necessary information required for Fraud Detection and Risk Management."